src/execution/mod.rs

```diff
@@ -61,68 +56,20 @@ pub struct ExecutionContext {
-    pub fn new(config: ExecutionConfig) -> Self {
-        let _ = &config; // avoid unused variable warning (it is used when some features are enabled)
-
-        let cfg = SessionConfig::default()
-            .with_batch_size(1)
-            .with_information_schema(true);
-
-        let runtime_env = RuntimeEnv::default();
-
-        #[cfg(feature = "s3")]
-        {
-            if let Some(object_store_config) = &config.object_store {
-                if let Some(s3_configs) = &object_store_config.s3 {
-                    info!("S3 configs exists");
-                    for s3_config in s3_configs {
-                        match s3_config.to_object_store() {
-                            Ok(object_store) => {
-                                info!("Created object store");
-                                if let Some(object_store_url) = s3_config.object_store_url() {
-                                    info!("Endpoint exists");
-                                    if let Ok(parsed_endpoint) = Url::parse(object_store_url) {
-                                        info!("Parsed endpoint");
-                                        runtime_env.register_object_store(
-                                            &parsed_endpoint,
-                                            Arc::new(object_store),
-                                        );
-                                        info!("Registered s3 object store");
-                                    }
-                                }
-                            }
-                            Err(e) => {
-                                log::error!("Error creating object store: {:?}", e);
-                            }
-                        }
-                    }
-                }
-            }
-        }
-
-        #[allow(unused_mut)] // used when deltalake is enabled
-        let mut state = SessionStateBuilder::new()
-            .with_default_features()
-            .with_runtime_env(runtime_env.into())
-            .with_config(cfg)
-            .build();
-
-        #[cfg(feature = "deltalake")]
-        {
-            state
-                .table_factories_mut()
-                .insert("DELTATABLE".to_string(), Arc::new(DeltaTableFactory {}));
+    pub fn try_new(config: &ExecutionConfig) -> Result<Self> {
+        let mut builder = DftSessionStateBuilder::new();
+        for extension in all_extensions() {
+            builder = extension.register(config, builder)?;
         }
 
-        {
-            let session_ctx = SessionContext::new_with_state(state);
+        let state = builder.build()?;
+        let session_ctx = SessionContext::new_with_state(state);
 
-            Self {
-                session_ctx,
-                #[cfg(feature = "flightsql")]
-                flightsql_client: Mutex::new(None),
-            }
-        }
+        Ok(Self {
+            session_ctx,
+            #[cfg(feature = "flightsql")]
```

Comment:
given the new organization i am wondering if `flightsql` client would be better a better fit in `app_execution`.  nothing for this PR, just thinking out loud.

Reply:
Yeah I agree the flightsql integration is still a little wonky. 

One thing I am wondering is rather than directly using the FlightSQL client, would it make more sense to have the UIs use an interface (aka trait) that takes SQL text and returns a `RecordBatchStream`? Then we could implement that for both ExecutionContext and flightsql 🤔 

It is more indirection but it would make it easy to use the same UI with different "backends" 🤔

Reply:
We might be able to - the only thing that gives me pause on that is the [statistics collection](https://github.com/datafusion-contrib/datafusion-dft/blob/main/src/execution/mod.rs#L184-L245) we do on the local executions to show additional statistics in the history tab.  right now its simple per query stats (elapsed time, and bytes scanned), but i had in mind (i.e. `ExecMetrics` in the link above) also showing summary stats per ExecutionPlan node (i have in mind displaying something like a heatmap table to show where the expensive parts of a query are).  It probably can be figured out how to work on a generic interface like that I just havent had time to think about it.

Reply:
We could maybe just stube that out / return no information for engines that don't support providing the info (actually in InfludDB 3.0 we return some of that info as grpc headers in the FlightSQL response, so it would be possible)

Anyhow, maybe we can do that when adding flightsql to the cli interface